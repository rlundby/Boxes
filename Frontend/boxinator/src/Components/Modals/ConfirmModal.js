import React from 'react';
import './Modals.css';

export default function ConfimModal({type, hideModal}) {


    if(type === 'success') {
        return(
            <div className='confirm-modal'>
                <h2>Success!</h2>
                <h3>The box has been added</h3>
                <button onClick={hideModal} className="button-modal">Continue</button>
            </div>
        )
    } else if(type === 'failed') {
        return(
            <div className='failed-modal'>
                <h2>Oh no - something went wrong!</h2>
                <h5>The box couldn't be added. Please make sure the Java REST Api is up and running!</h5>
                <button onClick={hideModal} className="button-modal failed">Ok :(</button>
            </div>
        )
    }
}