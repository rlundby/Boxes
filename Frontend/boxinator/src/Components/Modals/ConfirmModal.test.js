import React from 'react';
import {mount, shallow } from 'enzyme';

import ConfirmModal from './ConfirmModal';

const clickFn = jest.fn();

describe('ConfirmModal', () => {
    it('should fire function on button press', () => {
        const component = shallow(<ConfirmModal type="success" hideModal={clickFn}/>);
        component.find('button#button-toggle').simulate('click')
        expect(clickFn).toHaveBeenCalled();
    })

    it('should match snapshot', () => {
        const component = shallow(<ConfirmModal/>);
        expect(component).toMatchSnapshot();
    })

    it('should have className failed if post failed', () => {
        const component = mount(<ConfirmModal type="failed"/>);
        expect(component.childAt(0).hasClass('failed-modal')).toBe(true);
    })

    it('should not have className failed if post was successful', () => {
        const component = mount(<ConfirmModal type="success"/>);
        expect(component.childAt(0).hasClass('failed-modal')).toBe(false);
    })
})