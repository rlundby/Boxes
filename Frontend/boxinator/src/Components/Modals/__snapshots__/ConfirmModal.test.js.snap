// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal If failed, modal should have className failed 1`] = `""`;

exports[`ConfirmModal should fire function on button press 1`] = `""`;

exports[`ConfirmModal should match snapshot 1`] = `""`;
